src/model_config_tests/qa/test_access_esm1p5_config.py

```diff
@@ -0,0 +1,167 @@
+VALID_REALMS: set[str] = {"atmosphere", "land", "ocean", "ocnBgchm", "seaIce"}
+VALID_KEYWORDS: set[str] = {"global", "access-esm1.5"}
+VALID_NOMINAL_RESOLUTION: str = "100 km"
+VALID_REFERENCE: str = "https://doi.org/10.1071/ES19035"
+VALID_PREINDUSTRIAL_START: dict[str, int] = {"year": 101, "month": 1, "day": 1}
+VALID_HISTORICAL_START: dict[str, int] = {"year": 1850, "month": 1, "day": 1}
+VALID_RUNTIME: dict[str, int] = {"year": 1, "month": 0, "day": 0}
+VALID_RESTART_FREQ: str = "20YS"
+
+
+### Some functions to avoid copying assertion error text
+def error_field_nonexistence(field: str, file: str) -> str:
+    return f"Field '{field}' is null or does not exist in {file}."
+
+
+def error_field_incorrect(field: str, file: str, expected: Any) -> str:
+    return f"Field '{field}' in {file} is not expected value: {expected}"
+
+
+class AccessEsm1p5Branch:
+    """Use the naming patterns of the branch name to infer information of
+    the ACCESS-ESM1.5 config"""
+
+    def __init__(self, branch_name):
+        self.branch_name = branch_name
+        self.config_scenario = self.set_config_scenario()
+        self.config_modifiers = self.set_config_modifiers()
+
+    def set_config_scenario(self) -> str:
+        # Regex below is split into three sections:
+        # Config type start section: '^.+-' for 'release-', 'dev-'...
+        # Scenario section: '([^+]+)' for 'preindustrial', 'historical'...anything that isn't the '+' modifier sigil
+        # Modifiers end section: '(?:\+.+)*' any amount of '+modifer' sections
+        scenario_match = re.match(
+            r"^.+-(?P<scenario>[^+]+)(?:\+.+)*$", self.branch_name
+        )
+        if not scenario_match or "scenario" not in scenario_match.groupdict():
+            pytest.fail(
+                f"Could not find a scenario in the branch {self.branch_name}. "
+                + "Branches must be of the form 'type-scenario[+modifier...]'. "
+                + "See README.md for more information."
+            )
+        return scenario_match.group("scenario")
+
+    def set_config_modifiers(self) -> list[str]:
+        # Regex below is essentially 'give me the 'modifier' part in all the '+modifier's in the branch name'
+        return re.findall(r"\+([^+]+)", self.branch_name)
+
+
+@pytest.fixture(scope="class")
+def branch(control_path, target_branch):
+    branch_name = target_branch
+    if branch_name is None:
+        # Default to current branch name
+        branch_name = get_git_branch_name(control_path)
+        assert (
+            branch_name is not None
+        ), f"Failed getting git branch name of control path: {control_path}"
+        warnings.warn(
+            "Target branch is not specifed, defaulting to current git branch: "
+            f"{branch_name}. As some ACCESS-ESM1.5 tests infer information, "
+            "such as resolution, from the target branch name, some tests may "
+            "not be run. To set use --target-branch flag in pytest call"
```

Comment:
It isn't the case that we infer resolution from branch name for ESM configs. We may infer other things, but I think generally we'd be using the scenario and modifiers to determine optional tests.

Reply:
Updated in https://github.com/ACCESS-NRI/model-config-tests/pull/41/commits/c4d8a7d130b1a3764c34ba3c7cd0cbd7cb7971f8